server.go

```diff
@@ -1,10 +1,42 @@
 package dnscrypt
 
 import (
+	"net"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
 	"github.com/AdguardTeam/golibs/log"
 	"github.com/miekg/dns"
 )
 
+// default read timeout for all reads
+const readTimeout = 2 * time.Second
+
+// in case of TCP we only use readTimeout for the first read
+// then we start using tcpIdleTimeout
+const tcpIdleTimeout = 8 * time.Second
+
+// ServerDNSCrypt - interface for a DNSCrypt server
```

Comment:
1.  This is not an em-dash.
2.  English doesn't use em-dashes that way :-) .  Better:
    ```
    ServerDNSCrypt is a DNSCrypt server interface.